substrate-archive/src/actors/workers/storage_aggregator.rs

```diff
@@ -51,6 +55,18 @@ where
 		}
 		Ok(())
 	}
+
+	async fn handle_traces(&mut self, ctx: &mut Context<Self>) -> Result<()> {
+		let traces = std::mem::take(&mut self.traces);
+		if !traces.is_empty() {
+			log::info!("Inserting {} traces", traces.len());
+			for trace in traces {
```

Comment:
This is something that could be improved I think, however the reason I chose to do it this way is because each `trace` already contains probably 16-32 items (depending on what targets have been enabled) that are batch inserted, so I was worried about sending too much data at once if there were alot of traces. I think sending everything might be better though

Reply:
Worth investigating perhaps. If you agree, please add an issue to measure which is best.